app/main.py

```diff
@@ -1,3 +1,56 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(km=self.km + other)
+        return Distance(km=self.km + other.km)
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            self.km += other
+        else:
+            self.km += other.km
+
+        return self
+
+    def __mul__(self, number: int) -> Distance:
+        return Distance(km=self.km * number)
+
+    def __truediv__(self, number: int | float) -> Distance:
+        return Distance(km=round(self.km / number, 2))
```

Comment:
`other` can not be of type `Distance` in this method, because if you multiply km * km you get square km

Reply:
@YuriiKindrat but there `other` is only int #L27 and int/float in #L30